### PR TITLE
Firefox Account Authentication Support

### PR DESCRIPTION
**Pull request by derinb**

### CODE REVIEW

**Comment by derinb** [comment]
Andreas, I added check for Firefox Account authentication and disabled cookie-changed observer again.

Can you confirm it is working with the latest commit with a fresh profile?

**Comment by wagnerand** [comment]
I haven't tested this, but it looks good to me. Can you please cleanup the cookie-changed observer and squash the commits?

**Comment by derinb** [comment]
Or you can close this PR and I will upload only Authenticate Check with a proper PR name?

I have tested, it works with new profile and Firefox Accounts.

**Comment by wagnerand** [comment]
You can update the PR name. If not, I can. But since it's yours, you should be able to.

**Comment by derinb** [comment]
I just rebased but did not squash because there was 1 commit left. Hope it is OK.

**Comment by wagnerand** [comment]
Again, `Can you please cleanup the cookie-changed observer...`

There is no need to have dead code around. Cleaning up is a very important thing in such an old code base. It would be cool if we did more of it!

File: extension/modules/service.js [hunk 1]
```diff
@@ -1363,7 +1363,7 @@ var PersonaService = {
         if (uri.host != this.addonsHost)
             return;
 
-        if (endsWith(uri.filePath, "/users/logout") || aRequest.requestMethod == "POST" && endsWith(uri.filePath, "/users/login"))
+        if (endsWith(uri.filePath, "/users/logout") || aRequest.requestMethod == "POST" && endsWith(uri.filePath, "/users/login") || endsWith(uri.filePath, "/api/v3/accounts/authenticate/"))
```
Comment (wagnerand) [inline extension/modules/service.js]: You'd want to limit that added request to `POST` as well.

Comment (derinb) [inline extension/modules/service.js]: Andreas, that should keep there because some users may still use the old login system which uses POST.

**Comment by derinb** [comment]
Andreas, lets clean up the old code after we release restartless version and we make sure the login and custom personas bugs are fixed for all users.

I want to make the restartless transition as much painless as possible. I can remove the old code with another update XPI.

File: extension/modules/service.js (hunk 1 above)
Comment (wagnerand) [inline extension/modules/service.js]: I was talking about the url this patch adds. We only need ´POST` requests for that. Refreshing the favorites does not make sense if the request is a `GET` for example, as that means the login page has just loaded but not been submitted.

Comment (derinb) [inline extension/modules/service.js]: That authenticate is a GET request. If user validates, it is redirected to AMO. It happens after the POST request.

```
POST XHR https://api.accounts.firefox.com/v1/account/login [HTTP/1.1 200 OK 446ms]
POST XHR https://api.accounts.firefox.com/v1/certificate/sign [HTTP/1.1 200 OK 264ms]
POST XHR https://oauth.accounts.firefox.com/v1/authorization [HTTP/1.1 200 OK 338ms]
POST XHR https://accounts.firefox.com/metrics [HTTP/1.1 200 OK 270ms]
**GET https://addons.mozilla.org/api/v3/accounts/authenticate/ [HTTP/1.1 302 FOUND 436ms]**
```

Comment (wagnerand) [inline extension/modules/service.js]: Ah, sorry I misread that. I thought you were using the `authorization` request. But this one works as well.

**Comment by wagnerand** [comment]
Can you please use more descriptive commit titles? For example

`Add support for FxA login` or something.

You can change the commit title with just `git commit --amend`. Make sure nothing is staged, otherwise it will get amended to the commit as well.

**Comment by derinb** [comment]
Done.

**Comment by derinb** [comment]
Andreas, can you approve this so I can send another PR?

**Comment by wagnerand** [comment]
You can send PRs any time.

**Pull request merged**